File: DSA01016.cpp
#include <iostream>
#include <vector>

using namespace std;
vector<int> res;

void Try(int j, int n) {
    if (n == 0) {
        cout << "(";
        for (int k = 1; k < j; ++k) {
            cout << res[k];
            if (k < j - 1) {
                cout << " ";
            }
        }
        cout << ") ";
        return;
    }
    for (int i = min(res[j - 1], n); i > 0; --i) {
        res[j] = i;
        Try(j + 1, n - i);
    }
}

void function() {
    int n;
    cin >> n;
    res.resize(n + 1);
    res[0] = n;
    Try(1, n);
    cout << endl;
}

int main() {
    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        function();
    }
    return 0;
}

File: DSA01016.py
def Try(j, n):
    global res
    if n == 0:
        print("({})".format(" ".join(map(str,res[1:j]))), end = " ")
        return
    for i in range(min(res[j - 1], n), 0, -1):
        res[j] = i
        Try(j + 1, n - i)

def function():
    global res
    n = int(input())
    res = [0] * (n + 1)
    res[0] = n
    Try (1, n)
    print()



if __name__ == "__main__":
    for i in range(int(input())):
        function()